cmd/deployIntegrationArtifact_test.go

```diff
@@ -0,0 +1,113 @@
+package cmd
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	cpi "github.com/SAP/jenkins-library/pkg/cpi"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type deployIntegrationArtifactMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newDeployIntegrationArtifactTestsUtils() deployIntegrationArtifactMockUtils {
+	utils := deployIntegrationArtifactMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunDeployIntegrationArtifact(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfull Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Positive"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.NoError(t, err)
+	})
+
+	t.Run("Failed case of Integration Flow Deploy Test", func(t *testing.T) {
+		// init
+		config := deployIntegrationArtifactOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			Platform:               "cf",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "DeployIntegrationDesigntimeArtifact", StatusCode: 202,
+			ResponseBody: ``, TestType: "Negative"}
+
+		err := runDeployIntegrationArtifact(&config, nil, &httpClient)
+		// assert
+		assert.Error(t, err)
+	})
+
+}
+
+type httpMockCpis struct {
+	Method       string                  // is set during test execution
+	URL          string                  // is set before test execution
+	Header       map[string][]string     // is set before test execution
+	ResponseBody string                  // is set before test execution
+	Options      piperhttp.ClientOptions // is set during test
+	StatusCode   int                     // is set during test
+	CPIFunction  string                  // is set during test
+	TestType     string                  // is set during test
+}
+
+func (c *httpMockCpis) SetOptions(options piperhttp.ClientOptions) {
+	c.Options = options
+}
+
+func (c *httpMockCpis) SendRequest(method string, url string, r io.Reader, header http.Header, cookies []*http.Cookie) (*http.Response, error) {
+
+	c.Method = method
+	c.URL = url
+	if r != nil {
```

Comment:
Is that part really needed? I mean we are not doing anything with the content we read ...

Reply:
same is being retained by other httpmock impl used by some other steps such as GctsCreateRepository . so i retained it as well.

Reply:
That is strange. Maybe we should ask the colleagues for the reason why it is like it is.

Reply:
--> Reason for having this code here. We are closer to the behaviour of the real-world impl of SendRequest. The non-mocked instance - of course - consumes also `r`, which is basically the request body.